finagle-netty4/src/main/scala/com/twitter/finagle/netty4/ssl/server/Netty4ServerSslHandler.scala

```diff
@@ -8,9 +13,74 @@ import com.twitter.finagle.ssl.server.{
   SslServerSessionVerifier
 }
 import com.twitter.finagle.transport.Transport
-import com.twitter.finagle.Stack
-import io.netty.channel.{Channel, ChannelInitializer, ChannelPipeline}
-import io.netty.handler.ssl.SslHandler
+import com.twitter.util.{Future => TwitterFuture}
+import io.netty.buffer.ByteBufAllocator
+import io.netty.channel._
+import io.netty.handler.ssl._
+
+final case class SniSupport(mapping: SniSupport.ServerNameToContext)
```

Comment:
`String => TwitterFuture[Option[SslServerConfiguration]]` is more flexible, IMO. `Option[Map[String, SslServerConfiguration]]` implies the mapping is static and keys and certificates are available locally. For instance, it is not true in our case. We have keys and certificates stored in [Vault](https://www.vaultproject.io/) and new servers can be added at any time. That is why I also added https://github.com/twitter/finagle/pull/607/files/61cd8d956037e0efadcb186e48a54790b2c55802#diff-7be4dc145c09c1df2647a7ee1cc8bccd